core/optaplanner-constraint-streams-bavet/src/main/java/org/optaplanner/constraint/streams/bavet/common/index/ComparisonIndexer.java

```diff
@@ -16,72 +14,79 @@
 final class ComparisonIndexer<Tuple_ extends Tuple, Value_, Key_ extends Comparable<Key_>>
         implements Indexer<Tuple_, Value_> {
 
-    private final Function<IndexProperties, Key_> indexKeyFunction;
+    private final int indexKeyPosition;
     private final Supplier<Indexer<Tuple_, Value_>> downstreamIndexerSupplier;
-    private final BiPredicate<Key_, Key_> iterationStoppingCondition;
-    private final SortedMap<Key_, Indexer<Tuple_, Value_>> comparisonMap;
+    private final Comparator<Key_> keyComparator;
+    private final boolean isGteOrLte;
```

Comment:
(very soft) naming: how about "hasOrEquals"?

The current name itself didn't make me understand it's purpose, until I read how it's calculated.